packages/clay-card/src/index.tsx

```diff
@@ -4,101 +4,19 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-import AspectRatio from './AspectRatio';
-import Body from './Body';
-import Caption from './Caption';
-import classNames from 'classnames';
-import Context from './Context';
-import Description from './Description';
-import Group from './Group';
-import React from 'react';
-import {IContext} from './Context';
-
-type CardDisplayType = 'file' | 'image' | 'user';
-
-export interface ICardProps extends IContext {
-	displayType?: CardDisplayType;
-
-	/**
-	 * Flag that indicates if the card can be selectable.
-	 */
-	selectable?: boolean;
-}
-
-interface IProps
-	extends ICardProps,
-		React.BaseHTMLAttributes<HTMLAnchorElement | HTMLDivElement> {}
-
-const ClayCard: React.FunctionComponent<IProps> & {
-	AspectRatio: typeof AspectRatio;
-	Body: typeof Body;
-	Caption: typeof Caption;
-	Description: typeof Description;
-	Group: typeof Group;
-} = ({
-	children,
-	className,
-	displayType,
-	horizontal,
-	interactive,
-	selectable = false,
-	...otherProps
-}: IProps) => {
-	const Content: React.FunctionComponent<IProps> = ({children}) => (
-		<Context.Provider value={{horizontal, interactive}}>
-			{children}
-		</Context.Provider>
-	);
-
-	const isCardType = {
-		file: displayType === 'file',
-		image: displayType === 'image',
-		user: displayType === 'user',
-	};
-
-	const TagHeaderName = interactive ? 'a' : 'div';
-
-	const TagName = interactive ? 'span' : 'div';
-
-	return (
-		<TagHeaderName
-			{...otherProps}
-			className={classNames(className, {
-				card: !selectable,
-				'card-interactive card-interactive-primary card-type-template':
-					(horizontal && interactive) || interactive,
-				'card-type-asset':
-					!horizontal && !interactive && !(horizontal && interactive),
-				'card-type-directory form-check form-check-card form-check-middle-left':
-					selectable && horizontal,
-				'file-card': isCardType.file,
-				'form-check form-check-card form-check-top-left':
-					(selectable && isCardType.file) ||
-					isCardType.image ||
-					isCardType.user,
-				'image-card': isCardType.image,
-				'template-card': interactive && !horizontal,
-				'template-card-horizontal': horizontal && interactive,
-				'user-card': isCardType.user,
-			})}
-		>
-			{(selectable && !horizontal) ||
-			(selectable && isCardType.image) ||
-			isCardType.user ? (
-				<TagName className="card">
-					<Content>{children}</Content>
-				</TagName>
-			) : (
-				<Content>{children}</Content>
-			)}
-		</TagHeaderName>
-	);
+import ClayCard from './Card';
+import {ClayCardWithFile} from './CardWithFile';
+import {ClayCardWithFolder} from './CardWithFolder';
+import {ClayCardWithImage} from './CardWithImage';
+import {ClayCardWithNavigation} from './CardWithNavigation';
+import {ClayCardWithUser} from './CardWithUser';
+
+export {
```

Comment:
Not totally sure what you mean here. As we add high-level components I think its best to move the low-level entry point to its own file, and then import all components into `index` and export them appropriately. low-level as default and high-level as named exports.

Reply:
Sorry for not being clear. Currently in some components we have `index` with low-level components and the idea is to do the same as you are doing here. The `index` will only be for exporting the components, it will have no components inside.

Reply:
So I think it'd be fine to have an `index` page that exports a single component, or an `index` that is solely for exporting multiple components. However, we wouldn't want an `index` with both a defined component _and_ other high-level components. By doing this, we are able to avoid circular dependencies in the high-level components.

Overall, this approach is to avoid circular dependencies.